src/puppetlabs/puppetdb/queue.clj

```diff
@@ -183,14 +186,36 @@
   (update cmdref :attempts conj {:exception exception
                                  :time (kitchensink/timestamp)}))
 
+(defn store-in-stockpile [q command-stream received
+                          command version certname]
+  (try+
+   (stock/store q command-stream
+                (serialize-metadata received command version certname))
+
+   ;; Q: just delete the data for now?
+   (catch [:kind ::unable-to-commit] {:keys [stream-data]}
+     (-> "Unable to commit {0} v{1} for {2} received {3}; cleaning up"
```

Comment:
This will be double logged if you log it here and rethrow. Would probably be better to rethrow with this better error message as the exception message